server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1240,19 +1236,11 @@ public String source() {
             return source;
         }
 
-        public IndexReader reader() {
-            return searcher.getIndexReader();
-        }
-
         public DirectoryReader getDirectoryReader() {
```

Comment:
if we can move to a concrete reader then we can make this a hard cast to `ElasticsearchDirectoryReader`